src/rust/engine/fs/fs_util/src/main.rs

```diff
@@ -135,13 +141,22 @@ to this directory.",
 fn execute(top_match: clap::ArgMatches) -> Result<(), ExitError> {
   let store_dir = top_match.value_of("local-store-path").unwrap();
   let pool = Arc::new(ResettablePool::new("fsutil-pool-".to_string()));
-  let store = Arc::new(Store::local_only(store_dir, pool.clone()).map_err(|e| {
-    format!(
-      "Failed to open/create store for directory {}: {}",
-      store_dir,
-      e
-    )
-  })?);
+  let store = {
+    let store_result = match top_match.value_of("server-address") {
+      Some(cas_address) => {
+        Store::backfills_from_remote(store_dir, pool.clone(), cas_address.to_owned())
+      }
+      None => Store::local_only(store_dir, pool.clone()),
+    };
+    let store = store_result.map_err(|e| {
+      format!(
+        "Failed to open/create store for directory {}: {}",
```

Comment:
Should include the CAS address as well in case there is a connection issue.

... on that note, it occurs to me that we're going to have to do a _lot_ of thinking about failure modes of the remote store... which write-through cases it's safe to ignore errors for, timeouts, etc.

Reply:
A connection isn't actually attempted until you try to use the CAS, so the address will never be relevant to this failure.

And yeah, handling failure modes is going to be fun :)

Reply:
If you have time to put some thoughts as to our strategy in a ticket, that would be good. I suppose that since the remote execution API has explicit handling for signaling that a process doesn't have its inputs, we could do something like "best effort with a timeout" while backfilling, and only make remote errors fatal when attempting an execution.

Reply:
My plan is to only upload things when the process_execution is explicitly told that they're missing, rather than eagerly whenever we Store::store_bytes. We may want to add further heuristics in the future to eagerly upload things we suspect are likely to be missing from the remote CAS. Rather than a ticket, I'll add a descriptive comment and some TODOs in the next pull request I send which will implement CAS writes.

Right now, the behaviour is that we only check the remote CAS when we're trying to load a file which is missing locally, so any error ends up being a hard error which we should probably report. We may want to add some retries here, but I think that's a "deal with the problem when we come to it" kind of issue, rather than a "file a pre-emptive ticket about it" kind of thing...

Reply:
Reasonable.